src/controllers/main/main.ts

```diff
@@ -1626,13 +1626,13 @@ export class MainController extends EventEmitter {
     )
 
     // Update route status immediately, so that the UI quickly reflects the change
-    await Promise.all(
-      swapAndBridgeUserRequests.map(async (r) => {
-        await this.swapAndBridge.updateActiveRoute(r.meta.activeRouteId, {
-          routeStatus: 'in-progress'
-        })
+    // eslint-disable-next-line no-restricted-syntax
+    for (const r of swapAndBridgeUserRequests) {
+      // eslint-disable-next-line no-await-in-loop
+      await this.swapAndBridge.updateActiveRoute(r.meta.activeRouteId, {
+        routeStatus: 'in-progress'
       })
-    )
+    }
```

Comment:
What was the problem with Promise.all-ing these?

Reply:
The issue occurs because updateActiveRoute writes to this.activeRoutes. When multiple routes are updated in parallel, it causes a race condition, resulting in some routes failing to update properly